_posts/2016-05-11-android-testing-with-kotlin.md

```diff
@@ -0,0 +1,422 @@
+to have tons of utility and glue code just to prepare the test case, especially when you
+are testing our more complicated user flows. And to make the matters worse, in some cases the
+complexity of tests could easily exceed that of the main application. Those are exactly the
+cases where the verbosity of Java could get in the way. Every now and then it is just bloody
+convenient to open a file and loop over it's contents with one or two lines instead of
+couple of dozen LOCs that are required to do so in Java.
+
+In those situations a language like Kotlin can really help you to step up your game. The
+expressiveness and clean syntax, combined with nifty language idioms make it a huge
+improvement over the clumsiness that Java comes with. But enough of the small talk,
+let us see how it all works out in practice with a hands on example.
+
+## Plug Kotlin into your Android project
+
+> *For easier reference you can already grab complete source code of the example
+application from [this GitHub repository](https://github.com/GreenhouseCI/kotlin-tests).
+All code snippets to come will be extracted from there.*
+
+To start using Kotlin in your existing Android project you just have to add a few lines to your
+project's Gradle build file, declaring which version of Kotlin you'd like to use and
+which dependencies need to be additionally compiled. As far my experience goes,
+[the official documentation](https://kotlinlang.org/docs/reference/using-gradle.html),
+which seem to be great, will point you in the right direction. In any case, my `build.gradle`
+build script ended up being like this:
+
+```groovy
+apply plugin: 'com.android.application'
+apply plugin: 'kotlin-android'
+
+buildscript {
+    ext.kotlin_version = '1.0.+'
+    repositories {
+        jcenter()
+        mavenCentral()
+    }
+    dependencies {
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+    }
+}
+
+android {
+    ...
+}
+
+dependencies {
+    compile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    testCompile "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    testCompile "org.jetbrains.kotlin:kotlin-test-junit:$kotlin_version"
+    ...
+}
+```
+
+Let us tear down the extra lines. On the second line, right after Android plugin is applied,
+we inject Kotlin Gradle plugin that targets Android build model by
+`apply plugin: 'kotlin-android'`.
+To make that plugin available we need to declare it as a `buildscript` level dependency with
+
+```groovy
+dependencies {
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+}
+```
+where the preceding `ext.kotlin_version = '1.0.+'` just selects the newest version of Kotlin from
+1.0 series.
+
+Finally, under `dependencies` section we need to state that Kotlin _stdlib_ has to be compiled.
+And as we are going to use Kotlin especially for testing purposes, it needs to be compiled for
```

Comment:
~~especially~~ specifically?
"Especially" sounds like "a bunch of purposes, **especially** testing", while "specifically" is more like "just for this purpose"